experiment_code/hackrl/models/chaotic_dwarf.py

```diff
@@ -290,6 +292,17 @@ def initial_state(self, batch_size=1):
             for _ in range(2)
         )
 
+    def set_requires_grad(self, modules: Dict[str, bool]):
+        for name, requires_grad in modules.items():
+            for param in getattr(self, name).parameters():
+                param.requires_grad = requires_grad
+
+    def freeze(self, core: bool = True, actor: bool = True, critic: bool = True):
+        self.set_requires_grad({'core': not core, 'policy': not actor, 'baseline': not critic})
```

Comment:
Why you are skipping embeddings? Why not freeze all ane unfreeze whatever you want?